pwiz_tools/Skyline/Model/DocSettings/TransitionIonMobilityFiltering.cs

```diff
@@ -0,0 +1,1180 @@
+                            ion.PrecursorMz ?? mz, ion.Charge);
+                        if (!Equals(ionMobilityValue, result.IonMobility))
+                        {
+                            result = IonMobilityAndCCS.GetIonMobilityAndCCS(ionMobilityValue,
+                                result.CollisionalCrossSectionSqA, result.HighEnergyIonMobilityValueOffset);
+                        }
+                    }
+                }
+                return result;
+            }
+            else
+            {
+                return ionMobilities?.FirstOrDefault();
+            }
+        }
+
+        public List<IonMobilityAndCCS> GetIonMobilityInfoFromLibrary(LibKey ion)
+        {
+            // Locate this target in ion mobility library, if any
+            if (IonMobilityLibrary != null && !IonMobilityLibrary.IsNone)
+            {
+                var dict = IonMobilityLibrary.GetIonMobilityLibKeyMap();
+                if (dict != null && dict.TryGetValue(ion, out var imList))
+                {
+                    return imList;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Get the ion mobility for the ion, and the width of the window centered thereon
+        /// </summary>
+        public IonMobilityFilter GetIonMobilityFilter(LibKey ion, double mz,
+            IIonMobilityFunctionsProvider ionMobilityFunctionsProvider, double ionMobilityRangeMax)
+        {
+            var ionMobility = GetIonMobilityFilter(ion, mz, ionMobilityFunctionsProvider);
+            if (ionMobility != null)
+            {
+                double? ionMobilityWindowWidth;
+                if (ionMobility.IonMobility.HasValue)
+                {
+                    ionMobilityWindowWidth =
+                        FilterWindowWidthCalculator.WidthAt(ionMobility.IonMobility.Mobility.Value, ionMobilityRangeMax);
+                }
+                else
+                {
+                    ionMobilityWindowWidth = null;
+                }
+
+                return IonMobilityFilter.GetIonMobilityFilter(ionMobility, ionMobilityWindowWidth);
+            }
+            else
+            {
+                return null;
+            }
+
+        }
+
+        #region Implementation of IXmlSerializable
+
+        public enum ATTR
+        {
+            ion_mobility_library,
+            use_spectral_library_ion_mobility_values
+        }
+
+        public enum EL
+        {
+            ion_mobility_filtering
+        }
+
+        protected TransitionIonMobilityFiltering()
+        {
+
+        }
+
+        public XmlSchema GetSchema()
+        {
+            return null;
+        }
+
+
+        public static TransitionIonMobilityFiltering Deserialize(XmlReader reader)
+        {
+            return reader.Deserialize(new TransitionIonMobilityFiltering());
+        }
+
+
+        public void ReadXml(XmlReader reader)
+        {
+
+            var name = reader.Name;
```

Comment:
Local variable "name" never gets used.

Reply:
I always appreciate your eye for detail. Fixed.